chips/stm32f429zi/src/ethernet/mod.rs

```diff
@@ -0,0 +1,2328 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright 2023 OxidOS Automotive SRL
+//
+// Author: Ioan-Cristian CÎRSTEA <ioan.cirstea@oxidos.io>
+#![deny(missing_docs)]
+#![deny(dead_code)]
+#![deny(unused_imports)]
+//! Ethernet firmware for STM32F429ZI
+//!
+//! # Usage
+//!
+//! The Ethernet requires that AHB frequency must be at least 25MHz. Default boot configuration
+//! sets up a 16MHz clock frequency. Nucleo-F429ZI already contains code to setup the Ethernet.
+//! Uncomment code sections marked with ETHERNET to use this peripheral.
+
+use core::cell::Cell;
+
+use cortexm4::support::nop;
+use kernel::hil::ethernet::Configure;
+use kernel::hil::ethernet::EthernetAdapter;
+use kernel::hil::ethernet::EthernetAdapterClient;
+use kernel::hil::ethernet::EthernetSpeed;
+use kernel::hil::ethernet::MacAddress;
+use kernel::hil::ethernet::OperationMode;
+use kernel::platform::chip::ClockInterface;
+use kernel::utilities::cells::{NumericCellExt, OptionalCell, TakeCell};
+use kernel::utilities::registers::interfaces::{ReadWriteable, Readable, Writeable};
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
+use kernel::utilities::StaticRef;
+use kernel::ErrorCode;
+
+use crate::rcc;
+use crate::rcc::PeripheralClock;
+use crate::rcc::PeripheralClockType;
+
+mod transmit_descriptor;
+use crate::ethernet::transmit_descriptor::TransmitDescriptor;
+
+mod receive_descriptor;
+use crate::ethernet::receive_descriptor::ReceiveDescriptor;
+
+register_structs! {
+    /// Ethernet: media access control
+/// (MAC)
```

Comment:
We should fix the formatting in those register definitions.

Reply:
Fixed formatting.